### PR TITLE
fix: limit usernames to 20 characters

### PR DESCRIPTION
**Pull request by stephanegigandet**

### CODE REVIEW

File: lib/ProductOpener/Users.pm [hunk 1]
```diff
@@ -380,6 +380,9 @@ sub check_user_form($$$) {
 		elsif ($user_ref->{userid} !~ /^[a-z0-9]+[a-z0-9\-]*[a-z0-9]+$/) {
 			push @{$errors_ref}, $Lang{error_invalid_username}{$lang};
 		}
+		elsif (length($user_ref->{userid}) > 20) {
+			push @{$errors_ref}, $Lang{error_username_too_long}{$lang};
+		}
```
Comment (alexgarel) [inline lib/ProductOpener/Users.pm:385]: Being at it, can't we also add a check on username length < 60 characters for example (to avoid spam)

**Review by alexgarel — APPROVED** [review]
LGTM but I would like you to add a test on user name.

File: lib/ProductOpener/Users.pm (hunk 1 above)
Comment (stephanegigandet) [reply lib/ProductOpener/Users.pm:388]: @alexgarel I added the check.

**Review by alexgarel — APPROVED** [review]
LGTM + testé ok

**Pull request merged**